src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -581,6 +586,20 @@ public ProjectState WithAnalyzerReferences(IEnumerable<AnalyzerReference> analyz
             return With(projectInfo: ProjectInfo.WithAnalyzerReferences(analyzerReferences).WithVersion(Version.GetNewerVersion()));
         }
 
+        public ImmutableArray<ISourceGenerator> SourceGenerators
+        {
+            get
+            {
+                if (_lazySourceGenerators.IsDefault)
+                {
+                    var generators = AnalyzerReferences.SelectMany(a => a.GetGenerators(this.Language)).ToImmutableArray();
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazySourceGenerators, generators);
+                }
+
+                return _lazySourceGenerators;
+            }
+        }
```

Comment:
Is there any reason why we don't want to use the caching mechanism that analyzers/fixes/refactorings use?

Reply:
Which mechanism is that?

Reply:
@sharwell: I'm not seeing how that's helpful here. I don't want to reimplement any logic for how the actual types are discovered.